rust/arrow/src/datatypes.rs

```diff
@@ -156,6 +157,107 @@ make_type!(UInt64Type, u64, DataType::UInt64, 64, 0u64);
 make_type!(Float32Type, f32, DataType::Float32, 32, 0.0f32);
 make_type!(Float64Type, f64, DataType::Float64, 64, 0.0f64);
 
+macro_rules! make_temporal_type {
+    ($name:ident, $native_ty:ty, $data_ty:expr, $bit_width:expr, $default_val:expr) => {
+        pub struct $name {}
+
+        impl ArrowPrimitiveType for $name {
+            type Native = $native_ty;
+
+            fn get_data_type() -> DataType {
+                $data_ty
+            }
+
+            fn get_bit_width() -> usize {
+                $bit_width
+            }
+
+            fn default_value() -> Self::Native {
+                $default_val
+            }
+        }
+    };
+}
+
+make_temporal_type!(
+    TimestampSecondType,
+    i64,
+    DataType::Timestamp(TimeUnit::Second),
+    64,
+    0i64
+);
+make_temporal_type!(
+    TimestampMillisecondType,
```

Comment:
These names ended up becoming long. Would we prefer `TimestampType + TimestampMilliType + TimestampMicroType + TimestampNanoType` instead?

Reply:
Like mentioned above, if we agree to revamp the `DateType` for date, time, timestamp and interval, the name will be shorter.